mt-kahypar/partition/registries/register_refinement_algorithms.cpp

```diff
@@ -43,8 +43,8 @@
 #define REGISTER_FM_REFINER(id, refiner, t)                                                     \
   static kahypar::meta::Registrar<FMFactory> JOIN(register_ ## refiner, t)(                     \
     id,                                                                                         \
-    [](Hypergraph& hypergraph, const Context& context) -> IRefiner* {                           \
-    return new refiner(hypergraph, context);                                                    \
+    [](Hypergraph& hypergraph, const Context& context, const bool top_level) -> IRefiner* {     \
```

Comment:
The context has a member named ```type``` that indicates whether we are currently in the main direct k-way part or in initial partitioning.

Reply:
that was my fault, I told him to do so, because I did not remember that parameter. However, we should remove that top level flag on master

Reply:
Is updated